R/settings_utils.R

```diff
@@ -61,24 +61,26 @@ find_config <- function(filename) {
     dirname(filename)
   }
 
-  ## check for a file in the current directory
-  linter_config <- file.path(path, linter_file)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
-  }
-
-  ## next check for a file higher directories
-  linter_config <- find_config2(path)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
-  }
-
-  ## next check for a file in the user directory
-  # cf: rstudio@bc9b6a5 SessionRSConnect.R#L32
-  home_dir <- Sys.getenv("HOME", unset = "~")
-  linter_config <- file.path(home_dir, linter_file)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
+  # NB: This vector specifies a priority order for where to find the configs,
+  # i.e. the first location where a config exists is chosen and configs which
+  # may exist in subsequent directories are ignored
+  file_locations <- c(
+    # Current directory
+    file.path(path, linter_file),
+    # .github/linters directory
+    file.path(path, ".github", "linters", linter_file),
+    # Higher directories
+    find_config2(path),
+    # User directory
+    # cf: rstudio@bc9b6a5 SessionRSConnect.R#L32
+    file.path(Sys.getenv("HOME", unset = "~"), linter_file)
+  )
+
+  # Search through locations, return first valid result
+  for (loc in file_locations) {
+    if (isTRUE(file.exists(loc))) {
```

Comment:
Why the `isTRUE()` call? As far as I can tell `file.exists()` should always return exactly one logical value here so this code doesn’t need to guard against non-logical values or vectors of length > 1.

Reply:
Ugh, this comment (along with others that seem to be gone?!) was written months ago but was somehow part of a pending review without my noticing.